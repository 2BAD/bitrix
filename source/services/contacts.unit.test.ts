/* tslint-env jest */
// tslint:disable: no-expression-statement object-literal-sort-keys

import Contacts from './contacts'

const mockCall = jest.fn(() => Promise.resolve()) as any
const mockList = jest.fn(() => Promise.resolve()) as any
const contacts = Contacts({ call: mockCall, list: mockList })

const ID = '77'

describe('Contacts', () => {
  beforeEach(() => {
    mockCall.mockClear()
    mockList.mockClear()
  })

  it('should expose API methods', () => {
    expect(contacts).toMatchSnapshot()
  })

  describe('`create`', () => {
    it('should invoke `call`', async () => {
      await contacts.create({
        COMMENTS: 'Some comment'
      }, {
        REGISTER_SONET_EVENT: 'Y'
      })
      expect(mockCall.mock.calls).toMatchSnapshot()
    })

    it('should work without optional params', async () => {
      await contacts.create({
        COMMENTS: 'Some comment'
      })
      expect(mockCall.mock.calls).toMatchSnapshot()
    })
  })

  describe('`get`', () => {
    it('should invoke `call`', async () => {
      await contacts.get(ID)
      expect(mockCall.mock.calls).toMatchSnapshot()
    })
  })

  describe('`list`', () => {
    it('should invoke `list`', async () => {
      await contacts.list({
        start: 0,
        order: { COMMENTS: 'ASC' },
        filter: { '>PROBABILITY': 50 },
        select: ['*', 'UF_*']
      })
      expect(mockCall.mock.calls).toMatchSnapshot()
    })

    it('should work without optional params', async () => {
      await contacts.list()
      expect(mockCall.mock.calls).toMatchSnapshot()
    })
  })

  describe('`update`', () => {
    it('should invoke `call`', async () => {
      await contacts.update(ID, {
        COMMENTS: 'Some comment'
      }, {
        REGISTER_SONET_EVENT: 'Y'
      })
      expect(mockCall.mock.calls).toMatchSnapshot()
    })

    it('should work without optional params', async () => {
      await contacts.update(ID, {
        COMMENTS: 'Some comment'
      })
      expect(mockCall.mock.calls).toMatchSnapshot()
    })
  })
})
