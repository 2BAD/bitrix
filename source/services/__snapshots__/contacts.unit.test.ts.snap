// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts \`create\` should invoke \`call\` 1`] = `
Array [
  Array [
    "crm.contact.add",
    Object {
      "fields": Object {
        "COMMENTS": "Some comment",
      },
      "params": Object {
        "REGISTER_SONET_EVENT": "Y",
      },
    },
  ],
]
`;

exports[`Contacts \`get\` should invoke \`call\` 1`] = `
Array [
  Array [
    "crm.contact.get",
    Object {
      "id": "77",
    },
  ],
]
`;

exports[`Contacts \`list\` should invoke \`call\` 1`] = `Array []`;

exports[`Contacts \`update\` should invoke \`call\` 1`] = `
Array [
  Array [
    "crm.contact.update",
    Object {
      "fields": Object {
        "COMMENTS": "Some comment",
      },
      "id": "77",
      "params": Object {
        "REGISTER_SONET_EVENT": "Y",
      },
    },
  ],
]
`;

exports[`Contacts should expose API methods 1`] = `
Object {
  "create": [Function],
  "get": [Function],
  "list": [Function],
  "update": [Function],
}
`;
